src/MongoDB.Driver.Core/Core/Clusters/ClusterConnectionMode.cs

```diff
@@ -46,21 +49,23 @@ public enum ClusterConnectionMode
         Sharded
     }
 
-    internal static class ClusterConnectionModeExtensionMethods
+    /// <summary>
+    /// Determine whether to use legacy ConnectMode or DirectConnection.
+    /// </summary>
+    [Obsolete("Will be removed in a later version.")]
+    public enum ClusterConnectionModeSwitch
```

Comment:
Rename this enum to `ConnectionModeSwitch` and move to its own file.

Find all places where "clusterconnectionmodeswitch" is used and change to "connectionmodeswitch" (with whatever casing is appropriate).

The previous change will require moving some fields and properties around to maintain alphabetical order.

Note: we can consider whether in Core only the members can be called "clusterconnectionmodeswitch" to match the existing "clusterconnectionmode" members... but I'm inclined to use "connectionmodeswitch" everywhere even in Core.

Reply:
done